### PR TITLE
Move Articles::ScoreCalcJob to Sidekiq

### PR DESCRIPTION
**Pull request by atsmith813**
## What type of PR is this? (check all applicable)

- [x] Refactor
- [x] Optimization

## Description
This creates a new `Articles::ScoreCalcWorker` (spec included) and starts sending jobs to it. Once this is merged and deployed, I will open a second PR to remove the old job code and spec.

## Related Tickets & Documents
- #5305 

## Added to documentation?

- [x] no documentation needed

move_this_way_gif

### CODE REVIEW

File: app/models/article.rb [hunk 1]
```diff
@@ -64,7 +64,7 @@ class Article < ApplicationRecord
   before_save       :set_caches
   before_save       :fetch_video_duration
   before_save       :clean_data
-  after_save        :async_score_calc, if: :published
+  after_commit      :async_score_calc, if: :published, unless: :destroyed?
```
Comment (mstruve) [inline app/models/article.rb]: Can you really combine booleans like this? I think you might want to use a Proc here or possibly add a guard clause to the `async_score_calc` method below.

Comment (atsmith813) [reply app/models/article.rb]: Yep - it works 😀

https://guides.rubyonrails.org/v5.2/active_record_callbacks.html#multiple-conditions-for-callbacks

I'm happy to change it if we feel it's better with a Proc! What do you think?

Comment (mstruve) [reply app/models/article.rb]: TIL! IMO I think a single boolean would read better but it's not a huge deal for me.

Comment (citizen428) [reply app/models/article.rb]: I'm already not a big fan of conditional callbacks, so I'm with @mstruve here on not combining `if:` and `unless:` for a single callback.

File: app/workers/articles/score_calc_worker.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  class ScoreCalcWorker
+    include Sidekiq::Worker
+
+    sidekiq_options queue: :medium_priority
+
+    def perform(article_id)
+      article = Article.find_by(id: article_id)
+      return unless article
+
+      article.update_columns(score: article.reactions.sum(:points),
```
Comment (citizen428) [inline app/workers/articles/score_calc_worker.rb]: I really like my jobs not to know things about the app's business logic, so how about moving this out to some dedicated object like `ArticleScoreUpdater` or even just `Article_#update_score`?

Comment (atsmith813) [reply app/workers/articles/score_calc_worker.rb]: Updated to `Article_#update_score`.

File: app/models/article.rb (hunk 1 above)
Comment (atsmith813) [reply app/models/article.rb]: Updated to a guard clause.

**Pull request merged**